core/src/storage/collections/binary_heap/duplex_sync_chunk.rs

```diff
@@ -18,6 +18,18 @@
 //! of values in one cell (instead of the `SyncChunk` behavior of storing
 //! one value per cell). The intention is to reduce expensive fetch
 //! operations from storage.
+//!
+//! **NOTE** This wrapper is geared explicitly towards a binary tree
+//! structure -- the value which is stored at index `0` (the root) will
+//! always be stored in its own group with no other values in it. The
+//! intention is to be able to store child nodes paired together
+//! in a group.
```

Comment:
"... because for query operations you have to access both elements anyways. This allows to skip one expensive read for every accessed pair."

^--- The above is the rational why we actually do this.